### DISCUSSION TITLE
「重学TS 2.0 」TS 面试题：TypeScript 中 type 和 interface 之间有什么区别？

### DISCUSSION THREAD

**Discussion by semlinker** [semlinker/awesome-typescript#71] [General]
###  TypeScript 中 type 和 interface 之间有什么区别？

**Comment by wenkm** [comment]
先 `interface`，`interface` 不行再 `type`。因为 `interface` 能声明合并，而且……长得像好人。而 `type` 就不一样了，骚里骚气的。

**Comment by douhao1988** [comment]
### 相同点：
- 都可以描述 '对象' 或者 '函数' 
- 都允许拓展(`extends`)

### 不同点：
- `type` 可以声明基本类型，联合类型，元组
- type 可以使用 typeof 获取实例的类型进行赋值
- 多个相同的 interface 声明可以自动合并
- 使用 `interface` 描述 '数据结构'，使用 `type` 描述 '类型关系'
